cmd/exporter/main.go

```diff
@@ -43,135 +45,141 @@ var registeredResultMetrics []string
+	},
+		[]string{"app_uid", "engine_name", "kube_version"},
+	)
+
+	failedExperiments = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "c",
+		Subsystem: "engine",
+		Name:      "failed_experiments",
+		Help:      "Total number of failed experiments",
+	},
+		[]string{"app_uid", "engine_name", "kube_version"},
+	)
 )
 
 func contains(l []string, e string) bool {
-     for _, i := range l {
-         if i == e {
-             return true
-         }
-     }
-     return false
+	for _, i := range l {
+		if i == e {
+			return true
+		}
+	}
+	return false
 }
 
-func getEnv(key, fallback string)(string){
-        if value, ok := os.LookupEnv(key); ok {
-            return value
-        }
-        return fallback
+func getEnv(key, fallback string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return fallback
 }
 
-func metrics(cfg *rest.Config, cEngine string, aUUID string, aNS string){
-
-   for {
-            // Get the chaos metrics for the specified chaosengine 
-            expTotal, passTotal, failTotal, expMap, err := util.GetChaosMetrics(cfg, cEngine, aNS)
-            if err != nil {
-                //panic(err.Error())
-                log.Fatal("Unable to get metrics: ", err.Error())
-            }
-
-            // Define, register & set the dynamically obtained chaos metrics (experiment state)
-            for index, verdict := range expMap{
-                sanitizedExpName := strings.Replace(index, "-", "_", -1)
-                var (
-                    tmpExp = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-                        Namespace: "c",
-                        Subsystem: "exp",
-                        Name:      sanitizedExpName,
-                        Help: "",
-                    },
-                    []string{"app_uid", "engine_name"},
-                    )
-                )
-
-                if contains(registeredResultMetrics, sanitizedExpName) {
-                   prometheus.Unregister(tmpExp); prometheus.MustRegister(tmpExp)
-                   tmpExp.WithLabelValues(aUUID, cEngine).Set(verdict)
-                } else {
-                   prometheus.MustRegister(tmpExp)
-                   tmpExp.WithLabelValues(aUUID, cEngine).Set(verdict)
-                   registeredResultMetrics = append(registeredResultMetrics, sanitizedExpName)
-                }
-
-                // Set the fixed chaos metrics
-                experimentsTotal.WithLabelValues(aUUID, cEngine).Set(expTotal)
-                passedExperiments.WithLabelValues(aUUID, cEngine).Set(passTotal)
-                failedExperiments.WithLabelValues(aUUID, cEngine).Set(failTotal)
-            }
-
-            time.Sleep(1000 * time.Millisecond)
-   }
+func metrics(cfg *rest.Config, cEngine string, aUUID string, aNS string, kVer string) {
+
+	for {
+		// Get the chaos metrics for the specified chaosengine
+		expTotal, passTotal, failTotal, expMap, err := util.GetChaosMetrics(cfg, cEngine, aNS)
```

Comment:
We need to avoid util as a package name. GetChaosMetrics should not be exposed from util package.

Reply:
It is not good practice to have a function return more than 2 return values.

Reply:
This can also be done by using a struct, and would create abstraction for the users also. Let me have a talk with @ksatchit as well.

Reply:
This can also be done by using a struct, and would create abstraction for the users also. Let me have a talk with @ksatchit as well.